tests/qemu/test_qemu_img_utils.cpp

```diff
@@ -148,6 +165,29 @@ void test_image_conversion(const char* img_path, const char* expected_img_path,
     EXPECT_EQ(process_count, expected_final_process_count);
 }
 
+template <class Matcher>
+void test_image_amendment(const char* img_path,
```

Comment:
This is not how I would normally have done these tests, if from scratch. However, I didn't feel like modifying this whole file for this, so I just followed the existing approach.